### PR TITLE
[SHELL32] Make brsfolder.c C++ (as brsfolder.cpp)

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Modernize source code.
JIRA issue: [CORE-17340](https://jira.reactos.org/browse/CORE-17340)

## Proposed changes

- Move `dll/win32/shell32/wine/brsfolder.c` to `dll/win32/shell32/brsfolder.cpp`.
- Adapt to C++.

## TODO

- [x] Do build.
- [x] Do check.

### CODE REVIEW

File: dll/win32/shell32/brsfolder.cpp [hunk 1]
```diff
@@ -307,22 +285,22 @@ static HTREEITEM InsertTreeViewItem( browse_info *info, IShellFolder * lpsf,
 	if (!GetName(lpsf, pidl, SHGDN_NORMAL, szBuff))
 	    return NULL;
 
-	lptvid = SHAlloc( sizeof(TV_ITEMDATA) );
+    lptvid = (LPTV_ITEMDATA)SHAlloc( sizeof(TV_ITEMDATA) );
```
Comment (HBelusca) [inline dll/win32/shell32/brsfolder.cpp:288]: What's happening with the indentation?

File: dll/win32/shell32/brsfolder.cpp [hunk 2]
```diff
@@ -1168,34 +1145,24 @@ LPITEMIDLIST WINAPI SHBrowseForFolderA (LPBROWSEINFOA lpbi)
...
  */
-LPITEMIDLIST WINAPI SHBrowseForFolderW (LPBROWSEINFOW lpbi)
+EXTERN_C
+LPITEMIDLIST WINAPI SHBrowseForFolderW(LPBROWSEINFOW lpbi)
 {
-    browse_info info;
-    INT_PTR r;
-    HRESULT hr;
-    WORD wDlgId;
+    TRACE("%p\n", lpbi);
```
Comment (HBelusca) [inline dll/win32/shell32/brsfolder.cpp:1152]: I'm not convinced of the necessity of systematically adding such traces to every function.

**Comment by HBelusca** [comment]
I'm fine with the idea.

File: dll/win32/shell32/brsfolder.cpp (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/brsfolder.cpp]: Tab character is used in original source.
I will fix indentation later.

File: dll/win32/shell32/brsfolder.cpp (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/brsfolder.cpp]: Deleted.

**Comment by katahiromz** [comment]
Double checked. I have to go further.

**Pull request merged**